File: src/Mobile/piggy_flutter/lib/models/get_transactions_input.dart
import 'package:piggy_flutter/models/models.dart';

class GetTransactionsInput {
  GetTransactionsInput(
      {this.type,
      this.accountId,
      this.categoryId,
      this.startDate,
      this.endDate,
      this.groupBy}); // where the data is showing

  String? type;
  String? accountId;
  String? categoryId;
  DateTime? startDate;
  DateTime? endDate;
  String? query;
  TransactionsGroupBy? groupBy;
}

File: src/Web/aspnet-core/src/Piggyvault.Application/Piggy/Transactions/Dto/GetTransactionsInput.cs
using System;

namespace Piggyvault.Piggy.Transactions.Dto
{
    /// <summary>
    /// The get transactions input.
    /// </summary>
    public class GetTransactionsInput
    {
        /// <summary>
        /// Gets or sets the account id.
        /// </summary>
        public Guid? AccountId { get; set; }

        public Guid? CategoryId { get; set; }

        /// <summary>
        /// Gets or sets the end date.
        /// </summary>
        public DateTime? EndDate { get; set; }

        /// <summary>
        /// search
        /// </summary>
        public string Query { get; set; }

        /// <summary>
        /// Gets or sets the start date.
        /// </summary>
        public DateTime? StartDate { get; set; }

        /// <summary>
        /// Gets or sets the type.
        /// </summary>
        public string Type { get; set; }

        /// <summary>
        /// Gets or sets the user id.
        /// </summary>
        public long? UserId { get; set; }
    }
}